identity/src/main/java/com/stripe/android/identity/viewmodel/BottomSheetViewModel.kt

```diff
@@ -0,0 +1,45 @@
+package com.stripe.android.identity.viewmodel
+
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.lifecycle.ViewModel
+import com.stripe.android.identity.networking.models.VerificationPageStaticContentBottomSheetContent
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+
+/**
+ * ViewModel to control if a bottomsheet should show.
+ */
+@ExperimentalMaterialApi
+internal class BottomSheetViewModel : ViewModel() {
+
+    data class BottomSheetState(
+        val shouldShow: Boolean,
+        val content: VerificationPageStaticContentBottomSheetContent?
+    )
```

Comment:
we've the same case on connections, same bottom sheet with different contents. We follow the same approach as you're doing here, but worth mentioning there's some alpha support for bottom sheet destinations in Accompanist (so we might be able to migrate to something like that in the future) - https://google.github.io/accompanist/navigation-material/#bottom-sheet-destinations

Reply:
oh that's neat, thanks for the heads up. I also found the [ModalBottomSheet](https://developer.android.com/jetpack/compose/components/bottom-sheets) in material3 easier to use than the scafold